kong/plugins/prometheus/exporter.lua

```diff
@@ -39,21 +40,20 @@ local function init()
   prometheus = require("kong.plugins.prometheus.prometheus").init(shm, "kong_")
 
   -- global metrics
-  if kong_subsystem == "http" then
-    metrics.connections = prometheus:gauge("nginx_http_current_connections",
-      "Number of HTTP connections",
-      {"state"})
-  else
-    metrics.connections = prometheus:gauge("nginx_stream_current_connections",
-      "Number of Stream connections",
-      {"state"})
-  end
+  metrics.connections = prometheus:gauge("nginx_connections_total",
+    "Number of connections by subsystem",
```

Comment:
same, use `"kong_subsystem"` instead of `"subsystem"` in label, or change of them to `"subsystem"`.